IPNI.md

```diff
@@ -463,6 +463,11 @@ and uses the multihash portion of the CID only.
 * `cid` - _Required_. The default string representation of the Cid. Currently, Base32 is used for
   CIDv1 as the encoding for the multibase string and Base58 is used for CIDv0.
 
+##### Query Parameters
```

Comment:
What's the rationale for implementing this in the IPNI-specific API rather than as an extension to https://github.com/ipfs/specs/pull/337 (which IIUC is the thing meant to be used by external consumers of multiple routing systems such as kubo, helia, ...)? I guess it could be in both, but given that the semantics of different routing systems could be different it seems reasonable to me that IPNI would have a protocol more specifically tied to what it does, and a more generic API would cover plugging in additional routing systems.

Putting this in the non-IPNI API would also mean the IPNI API doesn't need to care about streaming results and such if it doesn't want to.

Reply:
Great point. 

There are two flavours of HTTP APIs: IPNI and the new HTTP delegated routing (ignoring the retired `/reframe` endpoints)

IPNI flavour predates the others. We need to maintain it for compatibility until, at the very least, HTTP delegated routing is stable.

Changing IPNI specs internally to what's needed has a faster turnaround and less resistance compared to the HTTP delegated routing. Therefore, the strategy here is to do both: try things in IPNI API first and see how they fit, then propose changes in HTTP delegated routing.

Reply:
> Changing IPNI specs internally to what's needed has a faster turnaround and less resistance compared to the HTTP delegated routing.

In this particular case I don't really understand why though. Basically you can choose to put the DHT lookup node in one of three configurations (note: I'm using domain names to roughly describe the layout, what the actual domain names are called, which group runs them, ... 🤷)
1. User -> cid.contact -> (IPNI-DB lookup + DHT lookup)
2. User -> routing.delegate.ipfs.io -> (DHT lookup + cid.contact)
3. User -> routing.delegate.ipfs.io -> (dht.delegate.ipfs.io + cid.contact)

Any of these options will get the job done. This approach is option 1. The reasons I personally prefer options 2 or 3 are:
1. It makes it harder to figure out what an IPNI endpoint does and what to do with it (i.e. my comments in this thread https://github.com/ipni/specs/pull/9#discussion_r1088968679)
2. It seems to cause tension when there is need for some capability in a routing system, that IPNI may/may not be interested in proxying (e.g. this IPNS comment https://github.com/ipfs-shipyard/someguy/issues/13#issuecomment-1406478765 and related comments in that issue)
   - There is a need for one (or a set) of delegated routing endpoints that low-resource clients (e.g. browsers) can leverage. Is that cid.contact, or something else, idk. But it tends to put pressure in this area.

Perhaps options 2/3 increase operational costs associated with caching in cloud infra compared to option 1, or there's some other good reason to go with option 1. If the IPNI team feels strongly that option 1 is the best way forward then 👍, I mostly just ask that my two concerns above be taken into consideration as we move things forward.

Reply:
No strong feelings here :)

Those options look very similar to me. I think no matter where we put this i'd like to point out:
* we will need some specification for it (i.e. the likes of this PR).
* this spec does not change the default cid.contact behaviour; the default response will include IPNI records only.
* considering cid.contact heavily utilises caching, and is already set up/running it seems like the path of least resistance to place this functionality in option 1 and take advantage of the caching.
* as you pointed out this will have cost saving benefits. 
* assuming the other endpoints listed above do not exist today, it'd also be the fastest path to make the service available.
* regarding [this thread](https://github.com/ipni/specs/pull/9#discussion_r1088968679), would [fa14d37](https://github.com/ipni/specs/pull/9/commits/fa14d372a34e5151dbb009f3b43f4733e7eb3e8d) address your concern?

Reply:
Yep, that addresses the concern thanks 🙏. As for where we should shuffle the query endpoints 🤷, to be tackled another day.

Reply:
Thank you @aschmahmann